examples/bigcommerce/lib/bigcommerce/bigcommerce.ts

```diff
@@ -0,0 +1,195 @@
+const STORE_NAME = process.env.BIGCOMMERCE_STORE_NAME
+const ACCESS_TOKEN = process.env.BIGCOMMERCE_ACCESS_TOKEN
+
+const getBearerToken = async () => {
+  const A_WEEK_FROM_NOW = Math.floor(
+    new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 7).getTime() / 1000,
+  )
+
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${process.env.BIGCOMMERCE_STORE_HASH}/v3/storefront/api-token`,
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Auth-Token': ACCESS_TOKEN ?? '',
+      },
+      body: JSON.stringify({
+        channel_id: 1,
+        expires_at: A_WEEK_FROM_NOW,
+        allowed_cors_origins: ['https://makeswift-example.mybigcommerce.com'],
+      }),
+    },
+  ).then(a => a.json())
+  return response
+}
+
+export async function getProducts() {
+  if (STORE_NAME == null) {
+    throw new Error('BIGCOMMERCE_STORE_NAME env variable is needed')
+  }
+
+  if (ACCESS_TOKEN == null) {
+    throw new Error('BIGCOMMERCE_ACCESS_TOKEN env variable is needed')
+  }
+
+  const bearerTokenResponse = await getBearerToken()
+
+  if (bearerTokenResponse.errors) {
```

Comment:
Shouldn't errors be handled by the `getBearerToken` function? It seems the function is leaking internal implementation details if it's returning errors. Specially since they're just being thrown by the caller. I would suggest checking for `response.ok` in `getBearerToken` and throwing an error there if the value is `false`.

Reply:
We seem to do that for this `getProducts` function (a bit later in the function) so it would also help with consistency